packages/block-editor/src/components/link-control/index.js

```diff
@@ -44,48 +43,33 @@ const MODE_EDIT = 'edit';
 
 function LinkControl( {
 	className,
-	currentLink,
-	currentSettings,
+	value,
+	settings,
 	fetchSearchSuggestions,
 	instanceId,
 	onClose = noop,
-	onChangeMode = noop,
-	onKeyDown = noop,
-	onKeyPress = noop,
-	onLinkChange = noop,
-	onSettingsChange = noop,
+	onChange = noop,
 } ) {
 	// State
 	const [ inputValue, setInputValue ] = useState( '' );
-	const [ isEditingLink, setIsEditingLink ] = useState( false );
-
-	// Effects
-	useEffect( () => {
```

Comment:
I was always in 2 minds about this. Sometimes I'm not clear about whether to manage dependencies manually or via hooks (like I did here). Now I assume instead of watching `currentLink` via the effect we're now switching the editing mode manually as and where required?

Reply:
yes, exactly, the problem is that the relationship between the value and the mode was not clear since you could have edit and view mode even if we have a value. It's not a 1-1 relationship, so I thought it's better to be explicit.